src/main/java/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -852,56 +852,67 @@ public void parseSetValueAction(ActionController source, Element e) {
         source.registerEventListener(event, action);
     }
 
+    public void parseSendAction(ActionController source, Element e) {
+        String event = this.getRequiredAttribute(e, "event");
+        String id = this.getRequiredAttribute(e, "submission");
+
+        SendAction action = new SendAction(id);
+        source.registerEventListener(event, action);
+    }
+
+    private String getRequiredAttribute(Element e, String attrName) {
+        String value = e.getAttributeValue(null, attrName);
+        if(value == null || value == "") {
+            throw new XFormParseException("Missing required attribute "+ attrName + " in element",
+                    e);
+        }
+        return value;
+    }
+
     private void parseSubmission(Element submission) {
         String id = submission.getAttributeValue(null, ID_ATTR);
 
-        //These two are always required
+        String resource = getRequiredAttribute(submission, "resource");
+        String targetref = getRequiredAttribute(submission, "targetref");
+
+        String ref = submission.getAttributeValue(null, "ref");
+
+        //For Validation Only
         String method = submission.getAttributeValue(null, "method");
-        String action = submission.getAttributeValue(null, "action");
 
-        SubmissionParser parser = new SubmissionParser();
-        for (SubmissionParser p : submissionParsers) {
-            if (p.matchesCustomMethod(method)) {
-                parser = p;
-            }
+        if(!("get".equals(method))) {

```

Comment:
OOC why are we only supporting `GET`

Reply:
POST just has much more complex semantics to implement, and we don't have an obvious use case for it.